### ISSUE TITLE
Recommend a format for serviceID 

### ISSUE THREAD

**Issue by marcoscaceres** [sysapps/telephony#203]
On the [mozilla lists](https://groups.google.com/forum/#!topic/mozilla.dev.webapi/bDOyYyKrkl0), the issue of the format for serviceID was raised with the following recommendation:

> we have to find a proper way to make sure this kind of ID is 100% unique cross different services (GSM, CDMA... and so on). Hsin-yi used to suggest maybe we can add a prefix or something like that to differentiate the purposes. For example, "CDMA:12345" and "GSM:12345" are different seviceIDs to identify different systems even if the "12345" part is the same by accident.

**Comment by zolkis** [comment]
The spec should only mandate uniqueness of service id's in the device / call history. It should not mandate any particular format: the format is the responsibility and freedom of implementations.

In the examples above, prepending with "GSM" and "CDMA" does not guarantee uniqueness, since there may be multiple GSM services in the system.

If there is a recommendation that makes sense globally, we could make it in a Note. But ultimately it is the responsibility of the implementations to handle it.

**Comment by marcoscaceres** [comment]
I agree that it should not mandate any format (i.e., no MUST). However, the question keeps coming up about what the RECOMMENDED identifiers one should use (I think I asked for guidance and now the Moz engineers also asked for guidance). We can add what you suggested to the spec to clarify this, but implementers do like having some kind of guidance about choosing ID format (as choosing the wrong kind, or adding one naively, can lead to privacy and security issues … so can be scary - particularly for telephony which already has a scary privacy/security aura around it).  

If it really does not matter, we can just say "use whatever you want" (e.g., a UUID or some kind of hash), but it really depends if these are only session identifiers, or long lived identifiers, etc.
